src/components/compound/DlSelect/DlSelect.vue

```diff
@@ -791,7 +796,7 @@ export default defineComponent({
             this.closeMenu()
         },
         selectOption(selectedOption: DlSelectOptionType) {
-            if (this.multiselect) {
+            if (this.multiselect || this.isReadonlyOption(selectedOption)) {
```

Comment:
@Yonatan-Mesika why would the readonly option be selectable even ?

Reply:
This method is called when there is a 'click' event (on the DlSelectOption), it isn't selectable.